llvm/tools/repo2obj/R2OELFSymbolTable.h

```diff
@@ -291,22 +295,23 @@ auto SymbolTable<ELFT>::insertSymbol(pstore::indirect_string const &Name,
 
 template <typename ELFT>
 auto SymbolTable<ELFT>::insertSymbol(pstore::indirect_string const &Name,
-                                     pstore::repo::relocation_type Type)
-    -> Value * {
-  auto SV = this->insertSymbol(Name, llvm::None);
+                                     pstore::repo::relocation_type Type,
+                                     bool IsWeak) -> Value * {
+  auto SV = this->insertSymbol(Name, llvm::None, IsWeak);
   SV->IsTLS = isTLSRelocation(Type);
   return SV;
 }
 
 template <typename ELFT>
 auto SymbolTable<ELFT>::insertSymbol(pstore::indirect_string const &Name,
-                                     llvm::Optional<SymbolTarget> const &Target)
-    -> Value * {
+                                     llvm::Optional<SymbolTarget> const &Target,
+                                     bool IsWeak) -> Value * {
   typename decltype(SymbolMap_)::iterator Pos;
   bool DidInsert;
   std::tie(Pos, DidInsert) = SymbolMap_.emplace(Name, Value{});
   if (DidInsert) {
     Pos->second = Value{Strings_.insert(Name), Target};
+    Pos->second.IsWeak = IsWeak;
```

Comment:
It might be cleaner to pass IsWeak to the Value ctor on line #311 rather than have it initialized to false and then re-assigned with the proper value here.

Reply:
Fixed, thanks.